README.md

```diff
@@ -1,31 +1,348 @@
+didn't have a `'things'` key, then `.any?` would raise a
+`NoMethodError`, interpreted as a non-match, and hyperion would move
+on to the next predicate.
+
+### Route classes
+
+In practice, you don't want to litter your code with `RestRoute.new`
+invocations. Here is a pattern for encapsulating the routes. It is the
+client-side analog of `routes.rb`.
+
+```ruby
+class CrudRoutes
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def read(id)
+    build(:get, id, response(message_type_for(@resource), 1, :json))
+  end
+
+  def create
+    build(:post, '', response(message_type_for(@resource), 1, :json), payload(:json))
+  end
+
+  ...
+end
+```
+
+You can easily imagine the rest of the routes and what the helper
+methods `build`, `message_type_for`, `response`, and `payload` look like.
+
+Use it like this:
+
+```ruby
+user_routes = CrudRoutes.new('users')
+Hyperion.request(user_routes.create, body: {name: 'joe', email: 'joe@schmoe.com'})
+# later...
+joe = Hyperion.request(user_routes.read(joes_id))
+```
+
+A few notes:
+
+_TODO: comment on testing_
+
+_TODO: comment on DRYing_
+
+
+### Configuration
 
 ```ruby
-gem 'hyperion'
+# configure hyperion
+Hyperion.configure do |config|
+  config.vendor_string = 'indigobio-ascent'  # becomes part of the Accept header
+end
 ```
 
-And then execute:
 
-    $ bundle
+## Superion
+
+Superion layers more convenience onto Hyperion by helping dispatch the
+response: rendering the response as an entity and dealing with errors.
+
+### Render and project
+
+```ruby
+require 'superion'
 
-Or install it yourself as:
+class UserGateway
+  include Superion
 
-    $ gem install hyperion
+  def find(id)
+    route = RestRoute.new(:get, "http://somesite.org/users/#{id}")
+    user = request(route, render: as_user)
+  end
+
+  def as_user
+    proc do |hash|
+      User.new(hash)
+    end
+  end
+end
+```
+
+On success, the "render" proc has a chance to transform the body
+(usually a `Hash`) into an internal representation (often an entity).
+
+After rendering, a "project" proc (the block) has a chance to project
+the rendered entity; for example, by choosing a subdocument or field.
+
+```ruby
+def user_names
+  route = RestRoute.new(:get, "http://somesite.org/users")
+  request(route, render: as_users) do |users|
+    users.map(&:name)
+  end
+end
+```
+
+### Result dispatch
+
+Superion has three levels of dispatching:
+
+- _core_,
+- _includer_, and
+- _request_.
+
+_TODO: these terms could use improvement. Suggestions?_
+
+They are distinguished by their scope. The core handler is built into
+superion. An includer handler affects all requests made by a
+particular class. A request handler affects only a particular request.
+
+When superion receives a response, it passes the result through the
+request, includer, and core handlers, in that order. The first handler
+to match wins, and no further handlers are tried. If no handler
+matches, then superion invokes the fallthrough handler.
```

Comment:
I think I'm going to get rid of the fallthrough handler altogether. It's only useful for 100s and 300s. If people really need a standard method for handling those, they can just call it at the end of `superion_handler`.